tests/test_text_cleaning.py

```diff
@@ -0,0 +1,413 @@
+        
+        for input_text, expected in test_cases:
+            with self.subTest(input_text=input_text):
+                result = self.parser._cleanup_remaining_parentheses(input_text)
+                self.assertEqual(result, expected)
+    
+    def test_cleanup_excessive_quotes(self):
+        """Test cleanup of excessive quotes."""
+        test_cases = [
+            ("'''bold''' text", "'bold' text"),
+            ("''italic'' text", "'italic' text"),
+            ("'''bold''' and ''italic''", "'bold' and 'italic'"),
+            ("Normal text", "Normal text"),  # No change
+            ("[['''link''']]", "['link']"),
+        ]
+        
+        for input_text, expected in test_cases:
+            with self.subTest(input_text=input_text):
+                result = self.parser._cleanup_excessive_quotes(input_text)
+                self.assertEqual(result, expected)
+    
+    def test_remove_list_markers(self):
+        """Test removal of list markers."""
+        text = """
+        * Item 1
+        ** Sub-item 1
+        *** Sub-sub-item
+        ** Sub-item 2
+        * Item 2
+        """
+        result = self.parser._remove_list_markers(text)
+        self.assertNotIn("* Item 1", result)
+        self.assertNotIn("** Sub-item 1", result)
+        self.assertNotIn("*** Sub-sub-item", result)
+        self.assertIn("Item 1", result)
+        self.assertIn("Sub-item 1", result)
+    
+    def test_cleanup_whitespace(self):
+        """Test whitespace cleanup."""
+        text = "Multiple    spaces    here.\n\n\nMultiple\n\n\nnewlines."
+        result = self.parser._cleanup_whitespace(text)
+        self.assertNotIn("    ", result)  # No multiple spaces
+        self.assertIn("Multiple spaces here", result)
+    
+    def test_apply_text_cleaning_regex_comprehensive(self):
+        """Test the comprehensive text cleaning method."""
+        text = """
+        == See also ==
+        * Link 1
+        * Link 2
+        
+        <gallery>
+        Image1.jpg
+        </gallery>
+        
+        Hello ({{lang|en|world}}) there.
+        
+        '''Bold''' and ''italic'' text.
+        
+        * List item
+        ** Sub-item
+        
+        Multiple    spaces.
+        """
+        
+        result = self.parser._apply_text_cleaning_regex(text)
+        
+        # Check that all cleaning operations were applied
+        self.assertNotIn("== See also ==", result)
+        self.assertNotIn("<gallery>", result)
+        self.assertNotIn("({{lang|en|world}})", result)
+        self.assertNotIn("'''Bold'''", result)
+        self.assertNotIn("* List item", result)
+        self.assertNotIn("    ", result)  # No multiple spaces
+        
+        # Check that content is preserved
+        self.assertIn("Hello", result)
+        self.assertIn("there", result)
+        self.assertIn("Bold", result)
+        self.assertIn("italic", result)
+        self.assertIn("List item", result)
+        self.assertIn("Sub-item", result)
```

Comment:
add assertions for expected cleaned output, not just the absence of unwanted patterns